Confus/Confus/Flag.h

```diff
@@ -1,69 +1,60 @@
 #pragma once
-namespace irr 
-{
-    class IrrlichtDevice;
-    namespace scene {
-        class IAnimatedMeshSceneNode;
-    }
-}
-
-namespace Confus 
-{
-	class Player;
-}
 
 namespace Confus 
 {
+class Player;
+class Collider;
 // Team Identifier
 /// <summary>
 /// The Team's Identifier. A player has a team, flag has a team, ui has a team, etc.
 /// </summary>
 enum class ETeamIdentifier 
 {
 	/// <summary>This object doesn't have a team identifier.</summary>
-	NONE,
+	None,
 	/// <summary>This object has a red team identifier.</summary>
-	TEAM_RED,
-	/// <summary>This object has a green team identifier.</summary>
-	TEAM_BLUE
+	TeamRed,
+	/// <summary>This object has a blue team identifier.</summary>
+	TeamBlue
 };
 enum class EFlagEnum
 { 
-    NONE,
-    FLAG_BASE,
-    FLAG_TAKEN,
-    FLAG_DROPPED 
+    None,
+    FlagBase,
+    FlagTaken,
+    FlagDropped 
 };
 
 	//Flag Class, every flag should have this class, contains info about a flag
 	/// Flag class with status and team id
 	class Flag 
     {
-    private:
-		irr::scene::IAnimatedMeshSceneNode* m_FlagNode;
+    private:		
 		irr::core::vector3df* m_StartPosition;
 		irr::core::vector3df* m_StartRotation;
 		EFlagEnum* m_FlagStatus;
         ETeamIdentifier* m_TeamIdentifier;
+        irr::scene::IMeshSceneNode* m_FlagNode;
+        Collider* m_Collider;
+        irr::scene::ISceneNode* m_FlagOldParent;
+
     public: 
-        /// <summary>
-        /// Flag class constructor
-        /// </summary>
-        /// <param name="a_SceneManager">The game's scene manager.</param>
-        /// <param name="a_TeamIdentifier">The team's identifier the flag should have.</param>
+        /// <summary> Flag class constructor. </summary>
+        /// <param name="a_SceneManager"> The game's scene manager. </param>
+        /// <param name="a_TeamIdentifier"> The team's identifier the flag should have. </param>
         Flag(irr::IrrlichtDevice* a_Device, ETeamIdentifier a_TeamIdentifier);
-        /// <summary>
-        /// Flag class destructor
-        /// </summary>
+        /// <summary> Flag class destructor </summary>
         ~Flag();
 		/// Capture Flag
 		void captureFlag(Player* a_PlayerObject);
         void setStartPosition(irr::core::vector3df a_Position);
         void setStartRotation(irr::core::vector3df a_Position);
 		void drop(Player* a_PlayerObject);
-		void returnToStartPos();
-		void score(Player* a_PlayerObject);;
+		void returnToStartPosition();
+		void score(Player* a_PlayerObject);
+        void setCollisionTriangleSelector(irr::scene::ISceneManager* a_SceneManager, irr::scene::ITriangleSelector* a_TriangleSelector);
 
+        irr::ILogger* MyLogger;
```

Comment:
Perhaps place this along with the other variables? It's hard to find this one the way it is